### PR TITLE
Adding Presto to Check

### PR DESCRIPTION
**Pull request by caiosba**
This PR adds Presto to Check as a Git Submodule. It also adds the service and its dependency ElasticMQ to Docker Compose. This way, we're able to run Presto locally as part of the Check suite. More details added as comments to the PR, see "Files changed".

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-dist: bionic
+dist: jammy
```
Comment (caiosba) [inline .travis.yml:1]: With `bionic` I was getting a weird error with `apt-get update` inside Presto's Docker:

```
Problem executing scripts APT::Update::Post-Invoke-Success
```

File: .travis.yml [hunk 2]
```diff
@@ -8,18 +8,19 @@ before_script:
   - sed -i 's/--abort-on-container-exit/-d/' bin/first-build.sh
   - TAB=$'\t'
   # For each submodule, go to a branch with the same name as this branch
-  - export FALLBACK_BRANCH=$([ "$TRAVIS_BRANCH" == "master" ] && echo "main" || echo "develop")
-  - git submodule foreach 'bash -c "git checkout develop ; git checkout $FALLBACK_BRANCH ; git checkout $TRAVIS_BRANCH ; exit 0"'
+  - export FALLBACK_BRANCH=$([ "$TRAVIS_BRANCH" == "master" ] && echo "main" && echo "master" || echo "develop")
+  - git submodule foreach 'bash -c "git checkout master ; git checkout develop ; git checkout $FALLBACK_BRANCH ; git checkout $TRAVIS_BRANCH ; exit 0"'
```
Comment (caiosba) [inline .travis.yml:12]: Fallback to `master` because there is no `develop` branch for Presto.

File: .travis.yml [hunk 3]
```diff
@@ -8,18 +8,19 @@ before_script:
...
   - ./bin/first-build.sh
   - until curl --silent -I -f --fail http://localhost:3000 ; do printf .; sleep 1; done
   - sleep 120
-  - if (( $(curl -s -o /dev/null -w "%{http_code}" http://localhost:3200)!= "200" ));  then exit 1; fi; # Pender
-  - if (( $(curl -s -o /dev/null -w "%{http_code}" http://localhost:3100)!= "200" ));  then exit 1; fi; # Alegre
-  - if (( $(curl -s -o /dev/null -w "%{http_code}" http://localhost:3333)!= "200" ));  then exit 1; fi; # Check Web
-  - if (( $(curl -s -o /dev/null -w "%{http_code}" http://localhost:3000)!= "200" ));  then exit 1; fi; # Check API
+  - if (( $(curl -s -o /dev/null -w "%{http_code}" http://localhost:3200)!= "200" )); then exit 1; fi; # Pender
+  - if (( $(curl -s -o /dev/null -w "%{http_code}" http://localhost:3100)!= "200" )); then exit 1; fi; # Alegre
+  - if (( $(curl -s -o /dev/null -w "%{http_code}" http://localhost:8000/ping)!= "200" )); then exit 1; fi; # Presto
```
Comment (caiosba) [inline .travis.yml:21]: Presto doesn't respond to `/`, so I added `/ping`.

File: bin/first-build.sh [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 # Go to the develop branch of each repository
-git submodule foreach bash -c 'git checkout develop'
+git submodule foreach bash -c 'git checkout develop || git checkout master'
```
Comment (caiosba) [inline bin/first-build.sh:4]: Presto has no `develop` branch.

File: docker-compose.yml [hunk 5]
```diff
@@ -167,6 +179,20 @@ services:
...
+      - "8000:8000"
+    volumes:
+      - "./presto:/app"
+    env_file:
+      - presto/.env_file
+    networks:
+      - dev
+    depends_on:
+      elasticmq:
+        condition: service_healthy
```
Comment (caiosba) [inline docker-compose.yml:195]: Another problem I had on CI was race conditions between Presto and ElasticMQ. Waiting for the service to be up fixed the problem. It was not enough to add just `depends_on: elasticmq`... I had to add a proper healthcheck that waits for the HTTP service to be actually up.

**Comment by skyemeedan** [comment]
will this merge be able to deploy to live? i.e. if SQS queues are not there yet, will presto still come up healthy enough that other services will work?

**Comment by caiosba** [comment]
> will this merge be able to deploy to live? i.e. if SQS queues are not there yet, will presto still come up healthy enough that other services will work?

Good question - This repository is never deployed to live, it's just used as the local development environment.

File: .travis.yml (hunk 1 above)
Comment (DGaffney) [reply .travis.yml:1]: I'm fine with that

**Review by DGaffney — APPROVED** [review]
Looks good to me!

**Pull request merged**